### PR TITLE
Enable OPENJ9_BUILD at makefile top level

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Enable `OPENJ9_BUILD` at `makefile` top level

Added `vendor_version.h`;
Removed `openj9_version.h`;
Removed individual `OPENJ9_BUILD` defines; 
Removed IBM vendor defines;
Refactored `openj9_version.c` and `openj9_version.h` to avoid multiple similar header files.

Reviewer @pshipton 
FYI: @DanHeidinga 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
A couple files contain the following, but since #1095 j9cfg.h.ftl does not. They should all be consistent.

```
#if defined(OPENJ9_BUILD)
/*The file collects version information in OpenJ9 build */
#include "openj9_version_info.h"
#endif /* OPENJ9_BUILD */
```

**Comment by JasonFengJ9** [comment]
Added `OPENJ9_BUILD` decoration for `#include "openj9_version_info.h"`.
Pls have another look.

**Comment by pshipton** [comment]
I wonder if we should remove the `#ifdef OPENJ9_BUILD` for all the `#include "openj9_version_info.h"`. Technically it is only there "temporarily" to support IBM builds, and shouldn't be used at all. 
I'd like to get @DanHeidinga opinions on this PR.

File: runtime/include/j9cfg.h.ftl [hunk 1]
```diff
@@ -33,7 +33,9 @@ extern "C" {
 #endif
 
 #include "omrcfg.h"
+#if defined(OPENJ9_BUILD)
```
Comment (DanHeidinga) [inline runtime/include/j9cfg.h.ftl]: if `openj9_version_info.h` is expected to always exist, then we shouldn't be guarding its inclusion with the `OPENJ9_BUILD` define.

I think this should be included unconditionally.

Comment (JasonFengJ9) [reply runtime/include/j9cfg.h.ftl]: Working on a PR to include `openj9_version_info.h` within `vendor_version.h` which is included unconditionally.

Comment (JasonFengJ9) [reply runtime/include/j9cfg.h.ftl]: PR has been updated, pls have another look.

**Comment by pshipton** [comment]
jenkins test all plinux

**Comment by pshipton** [comment]
Extended - JDK8 testing passed, the job failed due to #1114

**Pull request merged**